### PR TITLE
Rename Series: SeriesJson -> SeriesJsonPostgres; SeriesJsonB - > SeriesJson

### PR DESCRIPTION
**Pull request by thijs-obj**
## Main change: Renaming Json classes and dtypes
* `SeriesJson` -> `SeriesJsonPostgres`
  *  dtype: `json` -> `json_postgres`
* `SeriesJsonB` -> `SeriesJson`
  *  dtype: `jsonb` -> `json`

Now this makes things a bit confusing on Postgres, because `SeriesJson` handles the column type `jsonb` but has dtype `json`. But this has a very big advantage: `SeriesJson` and dtype `json` become the default json type. And the `SeriesJsonPg` becomes the exception. This means that on both BigQuery and Postgres users in general will use the same data type `SeriesJson`.

edit: 2022-05-23 16:21 CEST - changed naming from `SeriesJsonPG` to `SeriesJsonPostgres`

## Additional changes:
removed duplicated documentation on `SeriesJson.Json`

## Compatibility
Of course this change is not fully backwards compatible, so better to do it now than later, WDYT?

## Context
This PR is part of a series of PRs to get SeriesJson working for BigQuery:
1. **Rename Series:** https://github.com/objectiv/objectiv-analytics/pull/732
2. Move SeriesJson.Json class up one level: https://github.com/objectiv/objectiv-analytics/pull/733
3. Support JSON on BigQuery: https://github.com/objectiv/objectiv-analytics/pull/718

### CODE REVIEW

File: bach/bach/series/series_json.py [hunk 1]
```diff
@@ -132,85 +132,12 @@ class Json:
...
         class with accessor methods to json(b) type data columns.
         """
-        def __init__(self, series_object: 'SeriesJsonb'):
+        def __init__(self, series_object: 'SeriesJson'):
             self._series_object = series_object
 
         def __getitem__(self, key: Union[str, int, slice]):
             """
             Slice this jsonb database object in pythonic ways:
-
```
Comment (thijs-obj) [inline bach/bach/series/series_json.py:141]: This just duplicated the docstring of `SeriesJson`

File: bach/tests/functional/bach/test_data_and_utils.py [hunk 2]
```diff
@@ -218,7 +218,7 @@ def get_df_with_json_data(engine: Engine, dtype='string') -> DataFrame:
         df['dict_column'] = df.dict_column.astype(dtype)
         df['list_column'] = df.list_column.astype(dtype)
         df['mixed_column'] = df.mixed_column.astype(dtype)
-    return df
+    return df.materialize()
```
Comment (jansentom) [inline bach/tests/functional/bach/test_data_and_utils.py:221]: why is this required?

**Review by jansentom — APPROVED** [review]
LGTM, left one simple question.

File: bach/tests/functional/bach/test_data_and_utils.py (hunk 2 above)
Comment (thijs-obj) [reply bach/tests/functional/bach/test_data_and_utils.py:221]: This is not strictly needed. Just figured it's nice for the test to get a 'clean' dataframe

**Comment by thijs-obj** [comment]
In the latest commit ([3acc5a9](https://github.com/objectiv/objectiv-analytics/pull/732/commits/3acc5a9328e0d56e2251fa6189a0cd889ebcf640)), I changed the naming from `SeriesJsonPG` to `SeriesJsonPostgres` because of a comment in another PR: https://github.com/objectiv/objectiv-analytics/pull/733#discussion_r879186801

**Review by KathiaBarahona — APPROVED** [review]
lgtm! :rocket:
